shared/utils/yaml-to-shorthand.py

```diff
@@ -21,6 +21,14 @@
 from ssgcommon import open_yaml, required_yaml_key
 
 
+def add_warnings(element, warnings):
+    for warning_dict in warnings:
+        warning = add_sub_element(element, "warning", warning_dict.values()[0])
+        warning.set("category", warning_dict.keys()[0])
+
+    return element
```

Comment:
Don't you need to return the Etree element back to the function calling it?

Reply:
no?

Reply:
(in practice this is going to be pass by reference)

Reply:
Looks like the return is needed; otherwise, it is `None`.

Reply:
no, it is definitely not necessary, don't use the return value of this function, don't even assign it. the function will affect what you pass into it because python is pass by object or pass by ref or whatever you call it.

Reply:
`new_element = add_warning_elements(old_element, warnings)` will affect old_element without copying it! id(new_element) will be exactly the same as id(old_element).